spec/javascript/support/puppeteer/steps/supplementalApplicationSteps.js

```diff
@@ -0,0 +1,41 @@
+import utils from '../utils'
+
+const testStatusModalUpdate = async (page) => {
+  const COMMENT = 'This is a comment.'
+
+  // Wait for the status modal to appear
+  await page.waitForSelector('.form-modal_form_wrapper')
+
+  // Select a status different from the current one in the status modal
+  await page.click('.form-modal_form_wrapper .dropdown')
+  const newSelectedStatus = await page.$eval('.form-modal_form_wrapper .dropdown-menu li[aria-selected="false"] a', e => e.textContent)
+  await page.click('.form-modal_form_wrapper .dropdown-menu li[aria-selected="false"] a')
+
+  // Enter a comment into the status modal comment field
+  await page.type('#status-comment', COMMENT)
+
+  // Submit the status modal form
+  await page.click('.form-modal_form_wrapper button.primary')
+
+  // The page should enter a loading state while the status modal
+  // submit action is in progress
+  const pageHasLoadingState = await utils.isPresent(page, '.tabs-card-row > .tabs-card > .loading-panel.loading')
```

Comment:
is it possible that this could end up being really flakey because the timing might not actually work out that we'll catch the loading state?

Reply:
yeah i thought about that while putting this in, i'm not decided on it. on the one hand, it's nice to be able to test that the loading state appears when we want it to. on the other hand, timing isn't something we can always rely on. @fddayan what do you think?

Reply:
we've removed the loading state testing from this test and will test it in a future unit test